modules/rte_mis_state/rte_mis_state.module

```diff
@@ -293,3 +298,55 @@ function rte_mis_state_form_user_form_alter(&$form, &$form_state, $form_id) {
   }
 
 }
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function rte_mis_state_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  if ($view->id() == 'school_udise_code' && $view->current_display == 'page_1') {
+    // Getting the current user.
+    $user = \Drupal::currentUser();
+    $curr_user = User::load($user->id());
+    // Get the current user role.
+    $curr_user_role = $curr_user->getRoles();
+    // Check if they have district admin role assigned to them.
+    if (in_array('district_admin', $curr_user_role)) {
+      $definition = [
+        'table' => 'taxonomy_term__field_location',
+        'field' => 'entity_id',
+        'left_table' => 'taxonomy_term_field_data',
+        'left_field' => 'tid',
+        'type' => 'INNER',
+      ];
+      $join = Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      // Joining the taxonomy_term__field_location & taxonomy_term_field_data.
+      $query->addRelationship('taxonomy_term__field_location', $join, 'taxonomy_term_field_data');
+      // Get the current user's location detail.
+      $curr_user_location = $curr_user->get('field_location_details')->getString();
+      // Load the location vocabulary of the current user's location.
+      // Used to fetch the blocks under the current user's district.
+      $terms_tree = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadTree('location', $curr_user_location, 1, TRUE);
+      // Adding each block's id to an array.
+      foreach ($terms_tree as $term) {
+        $field_location = $term->id();
+        $field_location_arr[] = $field_location;
+      }
+      // Storing the value of array elements in string seperated by ','.
+      $value = implode(',', $field_location_arr);
+      // Adding an extra condition to the existing conditions.
+      $query->addWhereExpression(1, "taxonomy_term__field_location.field_location_target_id IN ( $value)");
+    }
+  }
+
+}
+
+/**
+ * Implements hook_user_login().
+ */
+function rte_mis_state_user_login(AccountInterface $account) {
+  // Check if the user has the 'district_admin' role.
+  if ($account->hasRole('district_admin')) {
+    // Invalidate the cache of the "school_udise_code" view.
+    Cache::invalidateTags(['config:views.view.school_udise_code']);
+  }
+}
```

Comment:
What will happen when a new UDISE is added for review and the district admin is already logged in?

Reply:
For this scenario, the data gets updated if the district_admin refreshes the page.